### PR TITLE
#979 cheatsheet integration with testing-tools

### PR DESCRIPTION
**Pull request by heheh111**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #979

### CODE REVIEW

File: Document/0x08-Testing-Tools.md [hunk 1]
```diff
@@ -4,11 +4,14 @@ To perform security testing different tools are available in order to be able to
 
 ### Mobile Application Security Testing Distributions
 
+- [Appie](https://manifestsecurity.com/appie/) - A portable software package for Android Pentesting and an awesome alternative to existing Virtual machines.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: Can you reformat the url? in ```[Appie](https://manifestsecurity.com/appie/ "appie")(eg. [title](url "title"))```  ?

Comment (commjoen) [reply Document/0x08-Testing-Tools.md]: And do this for all urls here?

File: Document/0x08-Testing-Tools.md [hunk 2]
```diff
@@ -4,11 +4,14 @@ To perform security testing different tools are available in order to be able to
 
 ### Mobile Application Security Testing Distributions
 
+- [Appie](https://manifestsecurity.com/appie/) - A portable software package for Android Pentesting and an awesome alternative to existing Virtual machines.
 - [Androl4b](https://github.com/sh4hin/Androl4b) - A Virtual Machine For Assessing Android applications, Reverse Engineering and Malware Analysis
 - [Android Tamer](https://androidtamer.com/) - Android Tamer is a Debian-based Virtual/Live Platform for Android Security professionals.
 - [AppUse](https://appsec-labs.com/AppUse/) - AppUse is a Virtual Machine developed by AppSec Labs.
+- [Mobisec](https://sourceforge.net/projects/mobisec/) - Mobile security testing live environment.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: Does this still work? seems 3 years old (can you verify it?)

File: Document/0x08-Testing-Tools.md [hunk 3]
```diff
@@ -4,11 +4,14 @@ To perform security testing different tools are available in order to be able to
...
 - [Android Tamer](https://androidtamer.com/) - Android Tamer is a Debian-based Virtual/Live Platform for Android Security professionals.
 - [AppUse](https://appsec-labs.com/AppUse/) - AppUse is a Virtual Machine developed by AppSec Labs.
+- [Mobisec](https://sourceforge.net/projects/mobisec/) - Mobile security testing live environment.
 - [Santoku](https://santoku-linux.com/) - Santoku is an OS and can be run outside a VM as a standalone operating system.
-- [Mobile Security Toolchain](https://github.com/xebia/mobilehacktools) - A project used to install many of the tools mentioned in this section both for Android and iOS at a machine running Mac OSX. The project installs the tools via Ansible
+- [Mobile Security Toolchain](https://github.com/xebia/mobilehacktools) - A project used to install many of the tools mentioned in this section both for Android and iOS at a machine running Mac OSX. The project installs the tools via Ansible.
+- [Vezir Project](https://github.com/oguzhantopgul/Vezir-Project) - Mobile Application Pentesting and Malware Analysis Environment.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: Again 3 years old... does it still work? Can you verify whether it works?

File: Document/0x08-Testing-Tools.md [hunk 4]
```diff
@@ -66,6 +70,20 @@ To perform security testing different tools are available in order to be able to
...
 - [Cydia Substrate Module: RootCoak Plus](https://github.com/devadvance/rootcloakplus) - Patch root checking for commonly known indications of root.
 - [Android-ssl-bypass](https://github.com/iSECPartners/android-ssl-bypass) - an Android debugging tool that can be used for bypassing SSL, even when certificate pinning is implemented, as well as other debugging tasks. The tool runs as an interactive console.
+- [Frida CodeShare](https://codeshare.frida.re/) - The Frida CodeShare project is comprised of developers from around the world working together with one goal - push Frida to its limits in new and innovative ways.
+
+#### Security Libraries
+
+- [PublicKey Pinning](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#Android) - Pinning in Android can be accomplished through a custom X509TrustManager. X509TrustManager should perform the customary X509 checks in addition to performing the pinning configuration.
+- [Android Pinning](https://github.com/moxie0/AndroidPinning) - A standalone library project for certificate pinning on Android.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: let's remove this as it is outdated. the way to do this is based on okhttp nowadays...

File: Document/0x08-Testing-Tools.md [hunk 5]
```diff
@@ -66,6 +70,20 @@ To perform security testing different tools are available in order to be able to
 - [Cydia Substrate Module: Android SSL Trust Killer](https://github.com/iSECPartners/Android-SSL-TrustKiller) - Blackbox tool to bypass SSL certificate pinning for most applications running on a device.
 - [Cydia Substrate Module: RootCoak Plus](https://github.com/devadvance/rootcloakplus) - Patch root checking for commonly known indications of root.
 - [Android-ssl-bypass](https://github.com/iSECPartners/android-ssl-bypass) - an Android debugging tool that can be used for bypassing SSL, even when certificate pinning is implemented, as well as other debugging tasks. The tool runs as an interactive console.
+- [Frida CodeShare](https://codeshare.frida.re/) - The Frida CodeShare project is comprised of developers from around the world working together with one goal - push Frida to its limits in new and innovative ways.
+
+#### Security Libraries
+
+- [PublicKey Pinning](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#Android) - Pinning in Android can be accomplished through a custom X509TrustManager. X509TrustManager should perform the customary X509 checks in addition to performing the pinning configuration.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: this one can be removed as well...

File: Document/0x08-Testing-Tools.md [hunk 6]
```diff
@@ -66,6 +70,20 @@ To perform security testing different tools are available in order to be able to
...
+
+- [PublicKey Pinning](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#Android) - Pinning in Android can be accomplished through a custom X509TrustManager. X509TrustManager should perform the customary X509 checks in addition to performing the pinning configuration.
+- [Android Pinning](https://github.com/moxie0/AndroidPinning) - A standalone library project for certificate pinning on Android.
+- [Java AES Crypto](https://github.com/tozny/java-aes-crypto) - A simple Android class for encrypting & decrypting strings, aiming to avoid the classic mistakes that most such classes suffer from.
+- [Proguard](https://www.guardsquare.com/en/products/proguard) - ProGuard is a free Java class file shrinker, optimizer, obfuscator, and preverifier. It detects and removes unused classes, fields, methods, and attributes.
+- [SQL Cipher](https://www.zetetic.net/sqlcipher/sqlcipher-for-android/) - SQLCipher is an open source extension to SQLite that provides transparent 256-bit AES encryption of database files.
+- [Secure Preferences](https://github.com/scottyab/secure-preferences) - Android Shared preference wrapper than encrypts the keys and values of Shared Preferences.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: There are some issues with this lib given https://github.com/scottyab/secure-preferences/issues .. can you verify them?

File: Document/0x08-Testing-Tools.md [hunk 7]
```diff
@@ -66,6 +70,20 @@ To perform security testing different tools are available in order to be able to
...
+- [Trusted Intents](https://github.com/guardianproject/TrustedIntents) - Library for flexible trusted interactions between Android apps.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: would this still make sense given the intents flow?

File: Document/0x08-Testing-Tools.md [hunk 8]
```diff
@@ -110,6 +135,11 @@ To perform security testing different tools are available in order to be able to
 - [iOS TrustMe](https://github.com/intrepidusgroup/trustme "iOS TrustMe") - Disable certificate trust checks on iOS devices.
 - [Xcon](http://cydia.saurik.com/package/com.n00neimp0rtant.xcon/ "Xcon") - A tool for bypassing Jailbreak detection.
 - [tsProtector](http://cydia.saurik.com/package/kr.typostudio.tsprotector8 "tsProtector 8") - Another tool for bypassing Jailbreak detection.
+- [Frida CodeShare](https://codeshare.frida.re/) - The Frida CodeShare project is comprised of developers from around the world working together with one goal - push Frida to its limits in new and innovative ways.
+
+#### Security Libraries
+- [PublicKey Pinning](https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning#iOS) - iOS pinning is performed through a NSURLConnectionDelegate. The delegate must implement connection:canAuthenticateAgainstProtectionSpace: and connection:didReceiveAuthenticationChallenge:. Within connection:didReceiveAuthenticationChallenge:, the delegate must call SecTrustEvaluate to perform customary X509 checks.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: not sure about this one...

File: Document/0x08-Testing-Tools.md (hunk 2 above)
Comment (heheh111) [reply Document/0x08-Testing-Tools.md]: The latest update was made on Jun 2, 2018, and from what I can see it seems to be up to date.

Comment (heheh111) [reply Document/0x08-Testing-Tools.md]: Latest update is from June 2018, and it looks like it is working as intended.

Comment (heheh111) [reply Document/0x08-Testing-Tools.md]: Up to date as of June 2018.

Comment (heheh111) [reply Document/0x08-Testing-Tools.md]: Working as of 31 Oct 2018

File: Document/0x08-Testing-Tools.md (hunk 6 above)
Comment (heheh111) [reply Document/0x08-Testing-Tools.md]: There are some bugs with this project. Not sure if it should be included so I am removing it.

**Comment by heheh111** [comment]
Resolved the changes. Please review this.

**Pull request merged**